### DISCUSSION TITLE
On the internationalization of dynamic data

### DISCUSSION THREAD

**Discussion by toffeeBlock** [lit/lit#4037] [Q&A]
I have a component as follows:
 
```
list.map(key => {
 return <components label=${key.label}></components>
})
```

where list is dynamically configured, possibly as follows:
```
const list = [
   {label: 'A'},
   {label: 'B'},
   {label: 'C'}
]
```

When I use ```@lit/localize``` for internationalization, this is what I write:
```
list.map(key => {
 return <components label=${msg(key.label)}></components>
})
```

When I generate the xlf file via ```lit-localize extract```, I get an error:

```Expected first argument to msg() to be a string or Lit template.```

I tried to use ```msg(html `${key.label}`) ``` to generate the xlf, and it doesn't give me an error, but it generates
```
<trans-unit id="haf63d34c8601dd41">
  <source><x id="0" equiv-text="${key.name}"/></source>
  <target><x id="0" equiv-text="${key.name}"/></target>
</trans-unit>

// After executing the lit-localize build, I get
const templates = {
      'h9519c78f95031572': html`${0}`
};
```

That is to say, no matter how I configure it, when switching languages, the label attribute can't be changed according to the current language, I would like to ask, is Lit currently unsupported this feature? Or is there something wrong with my writing style.

Currently I can only configure the xlf file manually according to the different label values, instead of using the lit-localize extract to generate it automatically!

I'm looking forward to a reply, thanks!

**Comment by augustjk** [comment]
Try this instead:
```ts
list.map(key => {
 return msg(html`<components label=${key.label}></components>`);
})
```

The `html` tag needs to wrap the entire html string to be understood.
